pkg/node/node.go

```diff
@@ -398,21 +329,32 @@ func createAPIServer(config NodeConfig, userKey *baccrypto.UserKey) (*publicapi.
 	return apiServer, nil
 }
 
-func createTransport(ctx context.Context, config NodeConfig) (*nats_transport.NATSTransport, error) {
-	transportLayer, err := nats_transport.NewNATSTransport(ctx, &nats_transport.NATSTransportConfig{
-		NodeID:                   config.NodeID,
-		Host:                     config.NetworkConfig.Host,
-		Port:                     config.NetworkConfig.Port,
-		AdvertisedAddress:        config.NetworkConfig.AdvertisedAddress,
-		AuthSecret:               config.NetworkConfig.AuthSecret,
-		Orchestrators:            config.NetworkConfig.Orchestrators,
-		StoreDir:                 config.NetworkConfig.StoreDir,
-		ClusterName:              config.NetworkConfig.ClusterName,
-		ClusterPort:              config.NetworkConfig.ClusterPort,
-		ClusterPeers:             config.NetworkConfig.ClusterPeers,
-		ClusterAdvertisedAddress: config.NetworkConfig.ClusterAdvertisedAddress,
-		IsRequesterNode:          config.IsRequesterNode,
-	})
+func createTransport(ctx context.Context, cfg NodeConfig) (*nats_transport.NATSTransport, error) {
+	storeDir, err := cfg.BacalhauConfig.NetworkTransportDir()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: revisit how we setup the transport layer for compute only, orchestrator only and hybrid nodes
+	config := &nats_transport.NATSTransportConfig{
+		NodeID:                   cfg.NodeID,
+		Host:                     cfg.BacalhauConfig.Orchestrator.Host,
+		Port:                     cfg.BacalhauConfig.Orchestrator.Port,
+		AdvertisedAddress:        cfg.BacalhauConfig.Orchestrator.Advertise,
+		AuthSecret:               cfg.BacalhauConfig.Orchestrator.Auth.Token,
```

Comment:
This single field, is used by both compute and orchestrator, right?

Reply:
it is. I am setting AuthSecret to be `cfg.BacalhauConfig.Compute.Auth.Token` if this is a compute only node.

```
if cfg.BacalhauConfig.Compute.Enabled && !cfg.BacalhauConfig.Orchestrator.Enabled {
		config.AuthSecret = cfg.BacalhauConfig.Compute.Auth.Token
	}

```